mesonbuild/compilers.py

```diff
@@ -1151,7 +1151,13 @@ def get_define(self, dname, prefix, env, extra_args, dependencies):
                 raise EnvironmentException('Could not get define {!r}'.format(dname))
         # Get the preprocessed value after the delimiter,
         # minus the extra newline at the end
-        return p.stdo.split(delim + '\n')[-1][:-1]
+        valuestr = p.stdo.split(delim + '\n')[-1]
+        # Sometimes GCC adds comment lines, ignore them.
+        for line in valuestr.split('\n'):
+            if len(line) > 0 and line[0] == '#':
+                continue
+            return line
```

Comment:
This will only return a single line of output while the earlier implementation returned multiline defines too.